kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -235,6 +236,9 @@ public LogSegment(
             () ->
                 Stream.concat(checkpoints.stream(), deltas.stream()).collect(Collectors.toList()));
 
+    // Make sure all delta log files are valid.
+    assertLogFilesBelongToTable(logPath, allFiles.get());
```

Comment:
So this is _not_ checking the log compactions! Let's clean things up:
- above `Member variable assignment` comment, invoke `assertLogFilesBelongToTable` with all of deltas + checkpoints + compactions (no need to save the complete list of all these 3 concatted to a variable)
- rename `allFiles` to `deltasAndCheckpoints` -- and make it not Lazy
- rename `allFilesReversed` tp `deltasAndCheckpointsReversed` -- keep it lazy
- rename `allFilesWithCompactionsReversed` to `deltasCheckpointsCompactionsReversed` -- keep it lazy